src/go/rpk/pkg/cli/cmd/container/common/common.go

```diff
@@ -219,7 +219,7 @@ func CreateNode(
 	}
 	metPort, err := nat.NewPort(
 		"tcp",
-		strconv.Itoa(config.Default().Redpanda.AdminApi.Port),
+		strconv.Itoa(config.Default().Redpanda.AdminApi[0].Port),
```

Comment:
i think RPK should talk to the 'named' connection or the one w/ TLS first.

Reply:
should we make the adminAPI a atype

````
type AdminAPI []Addresses

```

and then add receiver methods for saying Port() and Connect() which has embedded logic on what host to pick?

Reply:
I was fine with the `[0]` here because it is applied on the `Default()` configuration, which always only returns the default admin address: `{{0.0.0.0, 9644}}` - it doesn't select from a broader list. See my next comment for more.

Reply:
makes sense.

Reply:
but what if the default changes? It will be surprising to see this failed. I think my preference would be to name it and then retrieve it by name but not a blocker for me :)

Reply:
@alenkacz I think the default is unlikely to change here since this is in the `container` pkg, which generates the config on the spot, and the user can't modify it.